git-hooks/pre-commit

```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+#
+# An example hook script to verify what is about to be committed.
+# Called by "git commit" with no arguments.  The hook should
+# exit with non-zero status after issuing an appropriate message if
+# it wants to stop the commit.
+#
+# To enable this hook, rename this file to "pre-commit".
+
+if git rev-parse --verify HEAD >/dev/null 2>&1
+then
+	against=HEAD
+else
+	# Initial commit: diff against an empty tree object
+	against=4b825dc642cb6eb9a060e54bf8d69288fbee4904
+fi
+
+# If you want to allow non-ASCII filenames set this variable to true.
+# allownonascii=$(git config --bool hooks.allownonascii)
+
+# Redirect output to stderr.
+exec 1>&2
+
+DEBUG_FUNC=set_trace
+PRINT_FUNC=print
+NON_SAFE_PROTOCOL="http://"
+for invalid in $DEBUG_FUNC $PRINT_FUNC NON_SAFE_PROTOCOL;
```

Comment:
Why are you invalidating any use of set_trace or print? was this part of a previous git hook (couldn't see it in the source from 'dev' branch)?
I can see why this is a good idea, but it seems unrelated to the rest of the PR, and can be confusing. Maybe add a comment about this \ separate it to a different PR.